python/ray/autoscaler/v2/instance_manager/node_provider.py

```diff
@@ -80,37 +80,32 @@ def _filter_instances(
             filtered[instance_id] = instance
         return filtered
 
-    @override
-    def create_nodes(self, instance_type: InstanceType, count: int) -> List[Instance]:
-        result = self._node_launcher.launch_node(
-            self._config.get_node_config(instance_type.name),
+    def create_nodes(self, instance_type_name: str, count: int) -> List[Instance]:
+        created_nodes = self._node_launcher.launch_node(
+            self._config.get_raw_config_mutable(),
             count,
-            instance_type.name,
+            instance_type_name,
         )
         # TODO: we should handle failures where the instance type is
         # not available
-        if result:
+        if created_nodes:
             return [
                 self._get_instance(cloud_instance_id)
-                for cloud_instance_id in result.keys()
+                for cloud_instance_id in created_nodes.keys()
```

Comment:
Remind me again why we need to call get instance here? Is it because launch_node doesn't return all the necessariy info?